Add option to specify custom NNAPI serializer

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61025**
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434

Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

Commit:
Add option to specify custom NNAPI serializer

Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

Commit:
Update on "Add option to specify custom NNAPI serializer"


Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

Commit:
Update on "Add option to specify custom NNAPI serializer"


Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

[ghstack-poisoned]

dreiss commented on torch/backends/_nnapi/prepare.py:
```diff
@@ -75,14 +76,16 @@ def forward(self, args: List[torch.Tensor]) -> List[torch.Tensor]:
         inputs = [inputs]
 
-    ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = serialize_model(model, inputs)
+    if serializer:
+        ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = serializer.serialize_model(model, inputs)
+    else:
+        ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = _NnapiSerializer(config=None).serialize_model(model, inputs)
```
```
serializer = serializer or _NnapiSerializer(config=None)
ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = serializer.serialize_model(model, inputs)
```

Commit:
Update on "Add option to specify custom NNAPI serializer"


Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

[ghstack-poisoned]

Commit:
Update on "Add option to specify custom NNAPI serializer"


Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

Commit:
Update on "Add option to specify custom NNAPI serializer"


Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

Commit:
Update on "Add option to specify custom NNAPI serializer"


Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).
